lib/rouge/lexers/syzlang.rb

```diff
@@ -0,0 +1,315 @@
+        rule %r//, Text, :pop!
+      end
+
+      state :flags_list do
+        rule inline_spaces, Text
+        mixin :mixin_name
+        mixin :mixin_number
+        mixin :mixin_string
+        rule %r/,/, Punctuation, :space_break
+        rule %r//, Punctuation, :pop!
+      end
+
+      state :syscall_args do
+        rule spaces, Text
+        rule comment, Comment
+        rule id do
+          token Name
+          push :arg_type
+          push :space_break
+        end
+        rule %r/,/, Punctuation
+        rule %r/\)/, Punctuation, :pop!
+      end
+
+      state :syscall_retval do
+        mixin :mixin_name
+        rule %r//, Text, :pop!
+      end
+
+      state :syscall_mods do
+        rule %r/\(/, Punctuation, :term_list
+        rule %r//, Text, :pop!
+      end
+
+      state :struct_fields do
+        rule id do
+	  token Name
+          push :space_break
+          push :struct_field_mods
+          push :inline_break
+          push :arg_type
+          push :space_break
+        end
+        rule %r/[}\]]/, Punctuation, :pop!
+      end
+
+      state :struct_field_mods do
+        rule %r/\(/, Punctuation, :term_list
+        rule %r//, Text, :pop!
+      end
+
+      state :struct_mods do
+        rule %r/\[/, Punctuation, :term_list
+        rule %r//, Text, :pop!
+      end
+
+      state :type_name do
+        mixin :mixin_name
+        rule %r//, Text, :pop!
+      end
+
+      state :type_args do
+        rule %r/\[/, Punctuation, :type_args_list
+        rule %r//, Text, :pop!
+      end
+
+      state :type_args_list do
+        rule spaces, Text
+        rule comment, Comment
+        mixin :mixin_name
+        rule %r/,/, Punctuation
+        rule %r/\]/, Punctuation, :pop!
+      end
+
+      state :type_body do
+        rule %r/[{\[]/ do
+          token Punctuation
+          pop!
+          push :space_break
+          push :struct_mods
+          push :inline_break
+          push :struct_fields
+          push :space_break
+        end
+        rule %r// do
+          pop!
+          push :arg_type
+        end
+      end
+
+      state :root do
+        # Whitespace.
+        rule %r/#{spaces}/, Text
+
+        # Comments.
+        rule comment, Comment
+
+        # Includes.
+        rule %r/(include|incdir)/ do
+          token Keyword
+          push :include
+          push :space_break
+        end
+
+        # Defines.
+        rule %r/define/ do
+          token Keyword
+          push :define_exp
+          push :space_break
+          push :define_name
+          push :space_break
+        end
+
+        # Resources.
+        rule %r/resource/ do
+          token Keyword
+          push :resource_values
+          push :inline_break
+          push :resource_type
+          push :inline_break
+          push :resource_name
+          push :space_break
+        end
+
+        # Flags and strings.
+        rule %r/(#{id}|_)(#{spaces})(=)/ do |m|
+          if m[1] == ?_
```

Comment:
**question:** Could you help me to understand the use case for `?_`? Can we match a literal string `_`?

```suggestion
          if m[1] == "_"
```

Reply:
Indeed we can. Not sure why I decided to use this syntax. Done!